libeos-payg/manager.c

```diff
@@ -814,7 +817,7 @@ epg_manager_add_code (EpgProvider   *provider,
   g_array_sort (self->used_codes, used_codes_sort_cb);
 
   /* Extend the expiry time. */
-  extend_expiry_time (self, now_secs, period);
+  *time_added = extend_expiry_time (self, now_secs, period);
```

Comment:
There should be a `g_return_val_if_fail (time_added != NULL, FALSE)` at the top of this function.